### PR TITLE
Implement RandomAccessFile based storage

### PR DESCRIPTION
**Pull request by kuujo**
This PR implements the `DISK` based `StorageLevel` as a `RandomAccessFile` and adds a `MAPPED` storage level for memory mapped files.

### CODE REVIEW

**Comment by jhalterman** [comment]
Return of the MAPPED!

**Comment by jhalterman** [comment]
Change looks good, but maybe parameterized the Journal testing to run tests against different storage levels?

**Comment by kuujo** [comment]
Oops, I forgot there were level specific tests

File: tests/src/main/resources/logback.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
         </encoder>
     </appender>
 
-    <root level="${root.logging.level:-INFO}">
+    <root level="${root.logging.level:-TRACE}">
```
Comment (jhall11) [inline tests/src/main/resources/logback.xml]: Is this intentional?

**Comment by kuujo** [comment]
Fixed and fied

**Pull request merged**